### DISCUSSION TITLE
Limiting access to Github Organization

### DISCUSSION THREAD

**Discussion by oleks-popovych** [keycloak/keycloak#10006] [Q&A]
Hi guys,
great product, appreciate your work.

I have a quick question.
Is it possible to use GitHub as identity provider for keycloak, while allowing access only for members of my organization?

The idea behind this is to allow users to register, but limit them only to members on my organization on github.
I'd like to avoid checking who is allowed or create users for each request

**Comment by abstractj** [comment]
Hi @oleks-popovych, that's an interesting question. We do not provide that level of granularity, as you can [see at our documentation](https://www.keycloak.org/docs/latest/server_admin/index.html#_github). Although, it should be possible for you to implement [a custom identity provider](https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/services/src/main/java/org/keycloak/social/github/GitHubIdentityProvider.java) and introduce those checks.

If this is something that you would like to contribute to Keycloak, I'd suggest starting a conversation under the “Ideas” section.
